### ISSUE TITLE
Can't connect to the VPN, RNG errors

### ISSUE THREAD

**Issue by rKsanu2MMYvypWePtQWM** [angristan/openvpn-install#438]
Ubuntu server 18.04.2, all packages up-to-date
OpenVPN 2.4.4

Client can't connect.
What I've noticed is while generating a new client with the script, I get a few random generation errors:
```
Note: using Easy-RSA configuration from: ./vars

Using SSL: openssl OpenSSL 1.1.1  11 Sep 2018
Can't load /etc/openvpn/easy-rsa/pki/.rnd into RNG
140170845053376:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/etc/openvpn/easy-rsa/pki/.rnd
Generating an EC private key
writing new private key to '/etc/openvpn/easy-rsa/pki/private/testclient.key.PiXLGaC9UC'
-----
Using configuration from /etc/openvpn/easy-rsa/pki/safessl-easyrsa.cnf
Can't load /etc/openvpn/easy-rsa/pki/.rnd into RNG
140041032360384:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/etc/openvpn/easy-rsa/pki/.rnd
Check that the request matches the signature
Signature ok
The Subject's Distinguished Name is as follows
commonName            :ASN.1 12:'testclient'
Certificate is to be certified until Jun 10 14:13:12 2022 GMT (1080 days)

Write out database with 1 new entries
Data Base Updated

Client testclient added, the configuration file is available at /home/user/testclient.ovpn.
Download the .ovpn file and import it in your OpenVPN client.
```
Also I've noticed that the .ovpn files are a lot smaller than they used to be in older script releases. My old scripts with cipher AES-128-CBC, tls-cipher TLS-DHE-RSA-WITH-AES-128-GCM-SHA256 and tls-auth are 219 lines, while the file generated with this installation (default security settings) only gives files 76 lines long. Maybe these two things are related?

**Comment by angristan** [comment]
This is because we're using EC keys now, which are much smaller. The RNG error is unlikely a real issue

**Comment by rKsanu2MMYvypWePtQWM** [comment]
I'll try replicating my setup in a virtual machine for further testing. The system I'm trying to use is an old Core 2 Duo, maybe it lacks some kind of instruction that has become a requirement?

**Comment by angristan** [comment]
I don't think it's related

**Comment by rKsanu2MMYvypWePtQWM** [comment]
Changed from tls-crypt to tls-auth, started to work. WIth tls-crypt connection hangs and times out.

My client is also Ubuntu 18.04, with OpenVPN 2.4.4

**Comment by angristan** [comment]
This is really weird :thinking: Probably an edge case

**Comment by rKsanu2MMYvypWePtQWM** [comment]
Is there anything I can do to help debugging?

**Comment by angristan** [comment]
How do you connect on your client? Manually or using NetworkManager?

**Comment by rKsanu2MMYvypWePtQWM** [comment]
I just import the profile via Kubuntu's network settings, it probably uses networkmanager as framework.

**Comment by angristan** [comment]
Okay. I'm not sure but I think I remember that NetworkManager did not support tls-crypt. Maybe it does now, but with 18.04 you probably won't have it anyway

**Comment by angristan** [comment]
See https://github.com/angristan/openvpn-install/issues/312

**Comment by Papy-Serra** [comment]
About this issue, https://github.com/OpenVPN/easy-rsa/issues/261#issuecomment-444408090
I tested it, It seems working.
So how could I modify the file without being rewrite by your script ?

**Comment by angristan** [comment]
It should be fixed with easy-rsa 3.0.7

**Comment by Papy-Serra** [comment]
Yeah, I saw that and they also said that It should be release soon but... it was 5 months ago now :)
We just have to wait then.

**Comment by angristan** [comment]
Yes, afaik it's not really problematic

**Comment by Vorms** [comment]
Hello,
Me too I have an ubuntu desktop 18.04.
I am not a king in linux, but need openvpn.
I try a tuto to install easyRSA 3.0.5 and have the "Can't load /etc/openvpn/easy-rsa/pki/.rnd into RNG" issue
I installad the 3.0.6 and now read that the bug is solved in 3.0.7 but don't find the 3.0.7....

Can you tell me if it is possible to solve the 3.06 or wher be the 3.07 will be available ?
Best regards and thanks for your work ! 
Thierry Vorms vormsty@gmail.com

**Comment by nathan-io** [comment]
This issue [1][2] is still languishing and doesn't appear to be a priority for the Ubuntu maintainers.

In the meantime, you can bypass the problem by reconfiguring OpenVPN to use `tls-auth` instead of `tls-crypt`.

[1] https://bugs.launchpad.net/ubuntu/+source/network-manager/+bug/1749562
[2] https://bugs.launchpad.net/ubuntu/+source/network-manager-openvpn/+bug/1847144

**Comment by randshell** [comment]
> This issue [1][2] is still languishing and doesn't appear to be a priority for the Ubuntu maintainers.

You are linking a different issue. RNG one is fixed in EasyRSA 3.0.7 that we are already using.

Yours affects the Network-Manager so connecting from the command line works.

**Comment by nathan-io** [comment]
> RNG one is fixed in EasyRSA 3.0.7 that we are already using.

Thanks, glad to hear that.

> Yours affects the Network-Manager so connecting from the command line works.

Yes, connecting from CLI does work, but the connection is unusably slow. Not sure what the issue is there.

For now, I just reconfigured OpenVPN using all of the recommended encryption options, except for tls-auth over tls-crypt.
